vapi/rest/resource.go

```diff
@@ -57,6 +57,24 @@ func (r *Resource) WithParam(name string, value string) *Resource {
 	return r
 }
 
+// WithPathEncodedParam appends a parameter on the URL.RawQuery,
+// For special cases where URL Path-style encoding is needed
+func (r *Resource) WithPathEncodedParam(name string, value string) *Resource {
+	t := &url.URL{Path: value}
+	encodedValue := t.String()
+	t = &url.URL{Path: name}
+	encodedName := t.String()
+	// ParseQuery handles empty case, and we control access to query string so shouldn't encounter an error case
+	params, _ := url.ParseQuery(r.u.RawQuery)
```

Comment:
if we don't want to return an `error` here, we should still handle it and panic due to abnormal state instead of choking the error.